src/Serilog.Settings.Configuration/Settings/Configuration/ConfigurationReader.cs

```diff
@@ -154,27 +154,39 @@ Assembly[] LoadConfigurationAssemblies()
                 }
             }
 
-            foreach (var library in _dependencyContext.RuntimeLibraries)
+            foreach (var assemblyName in GetSerilogConfigurationAssemblies())
             {
-                if (library.Name != null && library.Name.ToLowerInvariant().Contains("serilog"))
-                {
-                    var assumedName = new AssemblyName(library.Name);
-                    var assumed = Assembly.Load(assumedName);
-                    if (assumed != null && !assemblies.ContainsKey(assumed.FullName))
-                        assemblies.Add(assumed.FullName, assumed);
-
-                    foreach (var assemblyRef in library.Assemblies)
-                    {
-                        var assembly = Assembly.Load(assemblyRef.Name);
-                        if (assembly != null && !assemblies.ContainsKey(assembly.FullName))
-                            assemblies.Add(assembly.FullName, assembly);
-                    }
-                }
+                var assumedName = new AssemblyName(assemblyName);
+                var assumed = Assembly.Load(assumedName);
+                if (assumed != null && !assemblies.ContainsKey(assumed.FullName))
+                    assemblies.Add(assumed.FullName, assumed);
             }
 
             return assemblies.Values.ToArray();
         }
 
+        string[] GetSerilogConfigurationAssemblies()
+        {
+            var query = Enumerable.Empty<string>();
+            var filter = new Func<string, bool>(name => name != null && name.ToLowerInvariant().Contains("serilog"));
+
+            if (_dependencyContext != null)
+            {
+                query = from lib in _dependencyContext.RuntimeLibraries where filter(lib.Name) select lib.Name;
+            }
+            else
+            {
+#if APPDOMAIN
+                query = from outputAssemblyPath in System.IO.Directory.GetFiles(AppDomain.CurrentDomain.BaseDirectory, "*.dll", System.IO.SearchOption.AllDirectories)
```

Comment:
`SearchOption.AllDirectories` might cause problems for apps that keep plug-ins in subfolders - often these contain distinct sets of assemblies that are loaded into different appdomains, meaning we could find multiple versions of DLLs of interest if this is specified.